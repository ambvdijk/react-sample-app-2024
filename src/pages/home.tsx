import React, { Fragment } from "react";

export function Home() {
  return (
    <Fragment>
      <h1>Homepage</h1>
      <nav>
        <ul>
          <li>
            <a href={`/page1`}>Page 1</a>
          </li>
          <li>
            <a href={`/page2`}>Page 2</a>
          </li>
        </ul>
      </nav>
    </Fragment>
  );
}
