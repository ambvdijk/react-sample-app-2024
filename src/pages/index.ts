export * from './home';
export * from './page1';
export * from './page2';