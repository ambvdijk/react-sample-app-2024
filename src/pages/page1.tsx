import { Link } from "react-router-dom";

export function Page1() {
  return (
    <section>
      <h1>This is page 1</h1>
      <Link to={`/`}>Go home</Link>
    </section>
  );
}
