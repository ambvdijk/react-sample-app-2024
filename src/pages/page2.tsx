import { Link } from "react-router-dom";

export function Page2() {
  return (
    <section>
      <h1>This is page 2</h1>
      <Link to={`/`}>Go home</Link>
    </section>
  );
}
