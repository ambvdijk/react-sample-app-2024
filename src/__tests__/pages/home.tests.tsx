import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Home } from "../../pages";

describe("Home page tests", () => {
  it("Should render correctly", () => {
    render(<Home />);
    expect(screen.getByRole("main")).toMatchSnapshot();
  });
  it("Should have 2 links", () => {
    render(<Home />);
    expect(screen.getByRole("nav").childElementCount).toBe(2);
  });
});
