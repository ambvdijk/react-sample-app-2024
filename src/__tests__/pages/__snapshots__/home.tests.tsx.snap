// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page tests Should render correctly 1`] = `
<main
  role="main"
>
  <h1>
    Homepage
  </h1>
  <nav>
    <ul
      role="nav"
    >
      <li>
        <a
          href="/page1"
        >
          Page 1
        </a>
      </li>
      <li>
        <a
          href="/page2"
        >
          Page 2
        </a>
      </li>
    </ul>
  </nav>
</main>
`;
